__init__.py

```diff
@@ -39,46 +41,175 @@ def wiki_image(pagetext):
+        auto_suggest (bool): wether or not to use autosuggest.
+
+    Returns:
+        PageMatch, PageDisambiguation or None
+    """
+    try:
+        # Use the version of Wikipedia appropriate to the request language
+        wiki.set_lang(lang_code)
+
+        # Fet wiki article titles. This comes back
+        # as a list.  I.e. "beans" returns ['beans',
+        #     'Beans, Beans the Music Fruit', 'Phaseolus vulgaris',
+        #     'Baked beans', 'Navy beans']
+        results = wiki.search(search, 5)
+        if len(results) == 0:
+            return None
+
+        return PageMatch(results[0], auto_suggest)
+
+    except wiki.exceptions.DisambiguationError as e:
+        # Test: "tell me about john"
+        return PageDisambiguation(e.options)
+
+
 class WikipediaSkill(MycroftSkill):
     def __init__(self):
         super(WikipediaSkill, self).__init__(name="WikipediaSkill")
 
     @intent_handler(IntentBuilder("").require("Wikipedia").
                     require("ArticleTitle"))
     def handle_wiki_query(self, message):
-        """ Extract what the user asked about and reply with info
-            from wikipedia.
+        """Extract what the user asked about and reply with info from wikipedia.
         """
-        # Talk to the user, as this can take a little time...
         search = message.data.get("ArticleTitle")
+        # Talk to the user, as this can take a little time...
         self.speak_dialog("searching", {"query": search})
+        self.handle_result(self.get_wiki_result(search))
+
+    def handle_result(self, result):
+        """Handle result depending on result type.
+
+        Speaks appropriate feedback to user depending of the result type.
+        Arguments:
+            result (object): wiki result object to handle.
+        """
+        if result is None:
+            self.respond_no_match()
+        elif isinstance(result, PageMatch):
+            self.respond_match(result)
+        elif isinstance(result, PageDisambiguation):
+            self.respond_disambiguation(result)
+
+    def respond_no_match(self):
+        """Answer no match found."""
+        self.speak_dialog("no entry found")
+
+    def respond_match(self, match):
+        """Read short summary to user."""
 
-        try:
-            self._lookup(search)
-        except wiki.PageError:
-            self._lookup(search, auto_suggest=False)
-        except Exception as e:
-            self.log.error("Error: {0}".format(e))
+        self.display_article(match)
+        # Remember context and speak results
+        self.set_context("wiki_article", json.dumps(match.serialize()))
+        self.set_context("spoken_lines", str(match.lines))
+        self.speak(match.summary)
+
+    def respond_disambiguation(self, disambiguation):
+        """Ask for which of the different matches should be used."""
+
+        options = (", ".join(disambiguation.options[:-1]) + " " +
+                   self.translate("or") + " " + disambiguation.options[-1])
```

Comment:
Could use the new `self.join_list(disambiguation.options, "or")` here

Reply:
That's what it was called looked around in util and didn't quite find what I was looking for. self.join_list sounds right. will try to use it.

Reply:
yeah it's not showing up in the docs so it's hard to find

Reply:
Updated